starlette/testclient.py

```diff
@@ -101,10 +122,14 @@ def __enter__(self) -> WebSocketTestSession:
             _: Future[None] = self.portal.start_task_soon(self._run)
             self.send({"type": "websocket.connect"})
             message = self.receive()
-            self._raise_on_close(message)
+            if message["type"] == "websocket.http.response.start":
+                self._handle_response(message)
+            else:
+                self._raise_on_close(message, reject=True)
         except Exception:
             self.exit_stack.close()
             raise
+        assert message["type"] == "websocket.accept"
```

Comment:
Is this here for type checking reasons?

Reply:
I don't think so, I think this is a genuine assert I put there, since this is a test client.
I can remove it if you like.